### ISSUE TITLE
Adding powers causing infinite loop

### ISSUE THREAD

**Issue by Drone4416** [kristianserrano/swade-item-tables-issue-tracker#15]
Whenever I or my player go to add spells to her character it creates two instances of the spell and then flashes in a third. It will also flash in a third anytime the power is clicked for any reason (editing, trying to drop the spell on the ground just to get it out of the inventory). This makes the character sheet unusable until I deactivate the module and delete the spell. The only odd thing I can think of on my end is that it is a custom arcane background.

**Comment by kristianserrano** [comment]
Thanks for reporting this. I'll look into this further. Out of curiosity, does this happen with only one user logged in?

**Comment by Drone4416** [comment]
Yes. Also did some testing if I have the spell pre-marked for a specific arcane background the third instance that is being made tries to be made under that set of power points. Also if I kill/crash the server it stops freaking out and will be left with 2-3 instances of the spell.

**Comment by kristianserrano** [comment]
Can you tell me which powers and from which sources these powers were pulled?

**Comment by kristianserrano** [comment]
Finally got a chance to test it, I was able to reproduce the infinite loop immediately. I think there's some code that I wrote for adding the New Powers Edge in the Character Builder both during creation and advances that makes some assumptions that I'll need to check for, I'll keep digging.

**Comment by kristianserrano** [comment]
@Drone4416 out of curiosity, did the Actor not have an Arcane Background when the power was dropped onto the character sheet?

**Comment by kristianserrano** [comment]
I think I've successfully fixed this with 1.21.9. Can you update and verify?

**Comment by Drone4416** [comment]
Alright so to start off with everything seems to be working correctly now but I'm going run down some of the things I had to do on my end in case it ever comes up for someone else or in case it helps with anything else. It didn't fix it initially but I found out that I hadn't added the AB to item tables because I have been retrofitting this into an active campaign. In addition even after adding it to item tables I had to re-add it to the sheet for it to stop doing the loop when I would add a spell.
